### PR TITLE
chore: refactor commands

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

closes https://github.com/celestiaorg/orchestrator-relayer/issues/277

This is best we can do. If we want to refactor further, we will need to use some dependency injection framework

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: cmd/qgb/keys/evm/evm.go [hunk 1]
```diff
@@ -76,14 +80,12 @@ func Add() *cobra.Command {
 			}(s, logger)
 
 			passphrase := config.EVMPassphrase
-			// if the passphrase is not specified as a flag, ask for it.
+			// if the new passphrase is not specified as a flag, ask for it.
```
Comment (SweeXordious) [inline cmd/qgb/keys/evm/evm.go:83]:
```suggestion
			// if the passphrase is not specified as a flag, ask for it.
```

**Comment by evan-forbes** [comment]
mind elaborating on why this is the best we can do (for posterity)?

**Review by evan-forbes — COMMENTED** [review]
could we also abstract out the connection the dht/all libp2p things? what about the queirers or the retrieer? I feel like there is still a lot of duplicate code around initialization that we _could_ (still optional!) abstract away, if that is the case, then there could still be a reason not to, but imo we should comment here why we don't want to so we can point to that discussion/decision in the future

**Comment by SweeXordious** [comment]
That's what I meant by dependency injection where we provide the configuration and the initialization is done automatically depending on the services that are needed.

For example:
https://github.com/celestiaorg/orchestrator-relayer/blob/a7311cc445d2032d58fb60d7b9258d95499b8668/cmd/qgb/orchestrator/cmd.go#L83-L110

It is not doing anything else aside from creating the querier and deferring its stop function. 

We could abstract a bit of it, but still that would be copying the code somewhere else and calling it, since not all commands need the same initialization. However, if we use dependency injection, we would just provide the configuration and the list of services needed, then they will be created automatically separately.

let me know if I'm missing something

**Comment by evan-forbes** [comment]
I guess I just don't considering an initialization function dependency injection. For example this code does not involve dependency injection, but would deduplicate logic. We could change the init of the queries in one spot, and it would get changed universally. The same could be applied for all things dht/libp2p afaiu

```go
func InitQueriers(config Config, logger Logger) (appQ, tmQ, stopFunc) {
	// creating tendermint querier
	tmQuerier := rpc.NewTmQuerier(config.tendermintRPC, logger)
	// creating the application querier
	appQuerier := rpc.NewAppQuerier(logger, config.celesGRPC, config.EncCfg)
	stopFunc := func() {
		err := appQuerier.Stop()
		if err != nil {
			logger.Error(err.Error())
		}
		err := tmQuerier.Stop()
		if err != nil {
			logger.Error(err.Error())
		}
	}
	return appQuerier, tmQuerier, stopFunc
}
```

**Comment by SweeXordious** [comment]
Aaah, I see what you mean. Okey, makes sense, I will update the PR. Thanks

File: cmd/qgb/common/helpers.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	if err != nil {
+		return nil, nil, err
+	}
+	stopFuncs = append(stopFuncs, func() error {
+		err = appQuerier.Stop()
+		if err != nil {
+			return err
+		}
+		return nil
+	})
```
Comment (evan-forbes) [inline cmd/qgb/common/helpers.go:59]: are trying to return this func to the user?

if so, then we have to accept a pointer to a slice or return one afaiu

**Review by evan-forbes — COMMENTED** [review]
nice!! thanks 

left one blocking question/comment, but then I think we're good to go!

File: cmd/qgb/common/helpers.go (hunk 2 above)
Comment (SweeXordious) [reply cmd/qgb/common/helpers.go:59]: we are returning it no? or you mean returning the pointer?

Comment (evan-forbes) [reply cmd/qgb/common/helpers.go:59]: ahh I think when I reviewed there were unpushed commits maybe? for whatever reason, I was reviewing 4bded402c0a0dcf3ecd299b08b8575eb7c5493b8, but you are correct, it was fixed!

**Pull request merged**